### PR TITLE
Lights api

### PR DESCRIPTION
**Pull request by plarios**
This PR corresponds to https://github.com/monoDriveIO/simulator-content/pull/111 and https://github.com/monoDriveIO/simulator/pull/305 and includes the configuration structs for the lights api as well as an example of how to use the api

### CODE REVIEW

**Review by devinaconley — CHANGES_REQUESTED** [review]
@plarios - should this be into `development`?

**Comment by stephendb** [comment]
Please update the PR comment to follow our current convention.
For new controls like this we need to know what the timing is for performance. Please attach to the PR comment.

1. Total Round trip time, and the break down:
- Time spent parsing json
- Time spent waiting on game thread
- Time spent changing lights

File: examples/cpp/lights/lights.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    std::string server0_ip = "127.0.0.1";
+    int server_port = 8999;   // This has to be 8999 this simulator is listening for connections on this port;
+
+    //Read JSON files in cpp_client/config directory
+    Configuration config(
+        "examples/config/simulator_straightaway.json",
+        "examples/config/weather.json",
+        "examples/config/scenario_multi_vehicle_straightaway.json"
+    );
+    config.simulator["map"] = "Straightaway5k_night";
```
Comment (bradleynull) [inline examples/cpp/lights/lights.cpp:21]: Suggest moving this to the scenario file and creating your own scenario for this example.

**Review by bradleynull — CHANGES_REQUESTED** [review]
@plarios Just a few minor comments.

File: examples/config/scenario_headlights.json [hunk 2]
```diff
@@ -0,0 +1,242 @@
+{
+  "trigger_boxes": [
```
Comment (bradleynull) [inline examples/config/scenario_headlights.json:2]: The other examples use config files in their own directory if they are not generic. I think this probably needs a refactor to move all config files to more logical spots, but for now can we move these into the lights directory?

**Review by bradleynull — CHANGES_REQUESTED** [review]
@plarios Looks good, just one comment.

**Review by bradleynull — APPROVED** [review]
@plarios This looks good to go after a quick fix to the pathing on the example that I pushed up. Approved!

**Pull request merged**